airbyte-workers/src/test/java/io/airbyte/workers/internal/state_aggregator/StateAggregatorTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.internal.state_aggregator;
+
+import static io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType.GLOBAL;
+import static io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType.LEGACY;
+import static io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType.STREAM;
+
+import com.google.common.collect.Lists;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.State;
+import io.airbyte.protocol.models.AirbyteGlobalState;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStreamState;
+import io.airbyte.protocol.models.StreamDescriptor;
+import java.util.List;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+public class StateAggregatorTest {
+
+  StateAggregator stateAggregator;
+
+  @BeforeEach
+  public void init() {
+    stateAggregator = new DefaultStateAggregator();
+  }
+
+  @ParameterizedTest
+  @EnumSource(AirbyteStateType.class)
+  public void testCantMixType(final AirbyteStateType stateType) {
+    final List<AirbyteStateType> allTypes = Lists.newArrayList(LEGACY, GLOBAL, STREAM);
```

Comment:
```suggestion
    final List<AirbyteStateType> allTypes = List.of(AirbyteStateType.values());
```
This way you don't need to hardcode the enum values in the test

Reply:
Done (switch it to Stream instead)